File: dgenesis/src/evaluate.c

/*
 *  GENESIS  Copyright (c) 1986, 1990 by John J. Grefenstette
 *  This program may be freely copied for educational
 *  and research purposes.  All other rights reserved.
 *
 *  file:	evaluate.c
 *
 *  purpose:	evaluate the current population by
 *		calling the user-defined function "eval"
 *
 *  modified:	13 feb 86
 *		12 nov 86: pass Length to Unpack()
 *		15 sep 90: handle floating pt representation,
 *		           change args to eval
 *		18 oct 93: consider newly arrived migrants (ECP)
 */

#include "extern.h"
extern double eval(); 


void Evaluate(nNewMigrants)
int nNewMigrants;
{
	register double performance;
	register int i;

	Trace("Evaluate entered");
	Dtrace("evaluate");

	for (i=0; i<Popsize; i++)
	{
		if ( New[i].Needs_evaluation )
		{
			Unpack(New[i].Gene, Bitstring, Length);
			if (Floatflag)
				FloatRep(Bitstring, Vector, Genes);
			New[i].Perf = eval (Bitstring, Length, Vector, Genes);
			performance = New[i].Perf;
			New[i].Needs_evaluation = 0;

			Trials++;
			Spin = 0;   /* we're making progress */

			if (Trials == 1)
				Best = performance;

			if (Savesize)  Savebest(i);

			if (BETTER(performance, Best))
				Best = performance;

			Onsum += performance;
			Offsum += Best;
			if (Dumpflag) Checkpoint(Ckptfile);
		}
	}

	for (i = Popsize; i < Popsize + nNewMigrants; i++){
		performance = New[i].Perf;
		if (BETTER(performance, Best))
			Best = performance;
		
		if (Savesize) Savebest(i);
	}
		
	Trace("Evaluate completed");
}


/*** end of file ***/

File: PARAGENESIS/evaluate.cs

/*
 *  file:	evaluate.cs
 *
 *  		Copyright (c) 1986, 1990 by John J. Grefenstette
 *
 *  purpose:	evaluate the current population by
 *		calling the user-defined function "eval"
 *
 *  modified:	13 feb 86
 *		12 nov 86: pass Length to Unpack()
 *		15 sep 90: handle floating pt representation,
 *		           change args to eval
 */

#include "Pextern.h"
extern double:physical Peval(); 


PEvaluate()
{

  Trace("Evaluate entered");
  Dtrace("evaluate");
  Time(0,"");

  with(physical) {
    where(Index < Popsize) {

	PUnpack(ParallelNew.Gene, PBitstring, Length);
	if (Floatflag)
	  PFloatRep(PBitstring, PVector, Genes);

	Trace("Starting Peval");
	ParallelNew.Perf = Peval (PBitstring, Length,
				  PVector, Genes);
	Trace("Finishing Peval");
	
	if (Trials == 0)
	  Best = (double)ParallelNew.Perf;
	
	Trials += (int:physical)1;
	Spin = 0;   /* we're making progress */
	
	if (Maxflag)
	  Best >?= ParallelNew.Perf;
	else
	  Best <?= ParallelNew.Perf;
	
	if (Savesize)  PSavebest(ParallelNew); 
	
	Onsum += ParallelNew.Perf;
	Offsum += Best * (+=(int:physical)1);
      }
  }
  

  if (Dumpflag) PCheckpoint(Ckptfile);

  Time(1,"Evaluate");
  Trace("Evaluate completed");
}


/** end of file **/
